packages/fxa-auth-server/lib/payments/capability.ts

```diff
@@ -378,38 +368,43 @@ export class CapabilityService {
     if (!customer || !customer.subscriptions!.data) {
       return [];
     }
-    const subscribedProducts = customer
+    const subscribedPrices = customer
       .subscriptions!.data.filter((sub) =>
         ACTIVE_SUBSCRIPTION_STATUSES.includes(sub.status)
       )
       .flatMap((sub) => sub.items.data)
-      .map(({ price: { product: productId } }) => productId as string);
-    return subscribedProducts;
+      .map(({ price: { id: priceId } }) => priceId as string);
+    return subscribedPrices;
   }
 
   /**
-   * Fetch the list of capabilities for the given product ids.
+   * Fetch the list of capabilities for the given price ids.
    */
-  private async productIdsToClientCapabilities(
-    subscribedProducts: string[]
+  private async priceIdsToClientCapabilities(
+    subscribedPrices: string[]
   ): Promise<ClientIdCapabilityMap> {
     const allCapabilities: Record<string, Set<string>> = {};
-
     // Run through all plans and collect capabilities for subscribed products
-    const plans = await this.stripeHelper.allAbbrevPlans();
-    for (const plan of plans) {
-      if (!subscribedProducts.includes(plan.product_id)) {
+    const prices = await this.stripeHelper.allAbbrevPlans();
+    for (const price of prices) {
+      if (!subscribedPrices.includes(price.plan_id)) {
         continue;
       }
-      const metadata = metadataFromPlan(plan);
-      const capabilityKeys = Object.keys(metadata).filter((key) =>
-        key.startsWith('capabilities')
-      );
-      for (const key of capabilityKeys) {
-        const capabilities = splitCapabilities((metadata as any)[key]);
-        const clientId = key === 'capabilities' ? '*' : key.split(':')[1];
-        for (const capability of capabilities) {
-          (allCapabilities[clientId] ??= new Set()).add(capability);
+      // Add the capabilities for this price and product
+      for (const metadata of [price.plan_metadata, price.product_metadata]) {
+        if (!metadata) {
+          continue;
+        }
+        const capabilityKeys = Object.keys(metadata).filter((key) =>
+          key.startsWith('capabilities')
+        );
+        for (const key of capabilityKeys) {
+          const capabilities = splitCapabilities((metadata as any)[key]);
+          const clientId =
+            key === 'capabilities' ? '*' : key.split(':')[1].trim();
+          for (const capability of capabilities) {
+            (allCapabilities[clientId] ??= new Set()).add(capability);
```

Comment:
Whew; this is a little challenging to understand; TIL the "nullish coalescing" operator in TS. I see it as: add the capability to the existing Set for this client ID if it exists, else create a new set and add the capability to the new set.

Reply:
Yup, this has been here since before this change. And its a slick little feature, but definitely something to add to ones repertoire. 😉